tests/pre-commit

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+#
+# Hook that executes the tests and format *.bz,*.bazel files automatically
+
+set -e
+
+function format_bazel() {
+  # Adapted from https://prettier.io/docs/en/precommit.html
+  FILES=$(git diff --cached --name-only --diff-filter=ACMR "*.bz" "*.bazel" | sed 's| |\\ |g')
+  [ -z "$FILES" ] && return 0
+
+  # Format all files (it's imprecise as we know only $FILES need modifications, but it'll do)
+  nix-shell --pure --command "bazel run //:buildifier-fix" || return 1
```

Comment:
That's risky. There have been bugs in buildifier in the past that caused lossy transformations or changed semantics. E.g. https://github.com/bazelbuild/buildtools/issues/645 or https://github.com/bazelbuild/buildtools/issues/766. I would generally recommend reviewing the changes performed by buildifier.